src/engine.c

```diff
@@ -3757,66 +3757,91 @@ int settle_legal(game *g, int who, int world, int mil_bonus, int mil_only,
 		/* Check for extra military */
 		if (o_ptr->code & P3_EXTRA_MILITARY)
 		{
+			/* Skip generic powers taken into account in total_military */
+			if (!(o_ptr->code & P3_CONDITIONAL_MILITARY)) continue;
+
+			/* Check for specificity */
```

Comment:
Not exactly the same:
- generic extra-military powers provide extra-military that can be used against all worlds and does not have to be paid
- the other powers provide extra-military that can either
  - only apply against a specific kind of worlds, e.g. against rebel
  - only apply if it is activated by a paiement (consume, discard, ...)
  or a combination of both.

So I think the two comments say something different. Would the solution of deleting the term "generic" of the first comment be acceptable to you?